### PR TITLE
PPCv2.0: "Guidance" checks with new structure

### PR DESCRIPTION
**Pull request by embarks**
## Context
See: https://github.com/google/web-stories-wp/pull/7980
- These checks were originally organized under "prepublish/guidance". That file structure has lost meaning due to visual changes to the PPC over time. But the bulk of the code comes from that implementation.

## Summary
- Refactored the "guidance" checks for the PPC to have their own component to fit into [the new structure](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=8378%3A1112).

## Relevant Technical Choices
- Did not include any video optimization check in this PR since a single video optimization does not work with this new structure, we need an option for bulk video optimization from the start. The blocked ticket with context is linked under TODO.
- This does not implement thumbnails or highlights, just the logic for the checks in this increment

## To-do

guidance
- [x] storyPagesCount
- [x] storyTitleLength
- [x] storyTooLittleText
- [x] pageTooMuchText
- [x] mediaElementResolution
- [x] videoElementLength
- [x] videoElementMissingPoster
- [x] ~videoElementOptimized~ this one will get done [in its own PR](https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/7309) as we have not yet implemented the bulk optimization feature.

## User-facing changes
none - behind a feature flag

### QA

- [x] This is a non-user-facing change and requires no QA

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

continues #7920

### CODE REVIEW

File: assets/src/edit-story/components/checklist/checks/imageElementResolution.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+const IMAGE_SIZE_HEIGHT = 1792;
+
+export function mediaElementResolution(element) {
+  switch (element.type) {
+    case 'image':
+      return imageElementResolution(element);
+    case 'gif':
+      return gifElementResolution(element);
+    default:
+      return () => false;
```
Comment (maxyinger) [inline assets/src/edit-story/components/checklist/checks/imageElementResolution.js:41]: should this just be?
```
default:
  return false;
```

File: assets/src/edit-story/components/checklist/checks/storyTitleLength.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { sprintf, _n } from '@web-stories-wp/i18n';
+
+/**
+ * Internal dependencies
+ */
```
Comment (maxyinger) [inline assets/src/edit-story/components/checklist/checks/storyTitleLength.js:23]: duplicate

File: assets/src/edit-story/components/checklist/checks/storyTooLittleText.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+function characterCountForPage(page) {
+  let characterCount = 0;
+  page.elements.forEach((element) => {
+    if (element.type === 'text') {
+      characterCount += stripHTML(element.content).length;
+    }
+  });
+  return characterCount;
+}
```
Comment (maxyinger) [inline assets/src/edit-story/components/checklist/checks/storyTooLittleText.js]: I think this is in the utils already rn

**Review by maxyinger — APPROVED** [review]
looks great. Loving the progress on this!

File: assets/src/edit-story/components/checklist/checks/storyTitleLength.js (hunk 2 above)
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/checks/storyTitleLength.js:23]:
```suggestion
```

File: assets/src/edit-story/components/checklist/checks/storyTooLittleText.js (hunk 3 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyTooLittleText.js]: you're right. I'm fixing this now!

**Comment by BrittanyIRL** [comment]
@googlebot I consent.

**Review by BrittanyIRL — APPROVED** [review]

**Pull request merged**